docs/docs/file-system-route-api.md

```diff
@@ -0,0 +1,299 @@
+
+```jsx:title=src/pages/index.js
+import React from "react"
+import { Link, graphql } from "gatsby"
+
+export default function HomePage(props) {
+  return (
+    <ul>
+      {props.data.allProduct.map(product => (
+        <li key={product.name}>
+          <Link to={product.productPath}>{product.name}</Link> (<Link to={product.discountPath}>Discount</Link>)
+        </li>
+      ))}
+    </ul>
+  )
+}
+
+export const query = graphql`
+  query {
+    allProduct {
+      name
+      productPath: gatsbyPath(filePath: "/products/{Product.name}")
+      discountPath: gatsbyPath(filePath: "/discounts/{Product.name}")
+    }
+  }
+}
+```
+
+By using [aliasing](/docs/graphql-reference/#aliasing) you can use `gatsbyPath` multiple times.
+
+## Example use cases
+
+Have a look at the [route-api example](https://github.com/gatsbyjs/gatsby/tree/master/examples/route-api/src/pages/products) for more detail.
+
+### Collection route + fallback
+
+By using a combination of a collection route with a client only route, you can create a great experience when a user tries to visit a URL from the collection route that doesn’t exist for the collection item. Consider these two file paths:
+
+- `src/pages/products/{Product.name}.js`
+- `src/pages/products/[name].js`
+
+If the user visits a product that wasn’t built from the data in your site, they will hit the client-only route, which can be used to show the user that the product doesn’t exist or load the data on the client.
+
+### Using one template for multiple routes
+
+By placing the template/view for your routes into a reusable component you can display the same information under different routes. Take this example:
+
+You want to display product information which is both accessible by name and SKU but has the same design. Create two file paths first:
+
+- `src/pages/products/{Product.name}.js`
+- `src/pages/products/{Product.meta__sku}.js`
+
+Create a view component at `src/view/product-view.js` that takes in a `product` prop. Use that component in your collection route, e.g.:
+
+```js:title=src/pages/products/{Product.name}.js
+import React from "react"
+import { graphql } from "gatsby"
+import ProductView from "../../views/product-view"
+
+function Product(props) {
+  const { product } = props.data
+  return <ProductView product={product} />
+}
+
+export default Product
+
+export const query = graphql`
+  query($id: String!) {
+    product(id: { eq: $id }) {
+      name
+      description
+      appearance
+      meta {
+        createdAt
+        id
+        sku
+      }
+    }
+  }
+`
+```
+
+### Purely client-only app
+
+If you want your Gatsby app to be 100% client-only you can create a file at `src/pages/[...].js` to catch all requests. See the [client-only-paths example](https://github.com/gatsbyjs/gatsby/tree/master/examples/client-only-paths) for more detail.
```

Comment:
```suggestion
If you want your Gatsby app to be 100% client-only routes, you can create a file at `src/pages/[...].js` to catch all requests. See the [client-only-paths example](https://github.com/gatsbyjs/gatsby/tree/master/examples/client-only-paths) for more detail.
```

Reply:
Why would someone want just client-only routes? Don't they lose the performanace benefit of running queries at build time?

Reply:
`client-only` makes more sense in this context :) 

Instead of using Create React App people sometimes prefer our ecosystem of plugins, themes, etc. which also works with purely client-only apps

Reply:
Got it! Also thanks for catching that, I updated it to just be adding the comma :)